processor/rollup.base.config.js

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import nodeResolve from "rollup-plugin-node-resolve";
+import multiEntry from "rollup-plugin-multi-entry";
+import cjs from "rollup-plugin-commonjs";
+import json from "rollup-plugin-json";
+import replace from "rollup-plugin-replace";
+import { uglify } from "rollup-plugin-uglify";
+import sourcemaps from "rollup-plugin-sourcemaps";
+
+import path from "path";
+
+const pkg = require("./package.json");
+const depNames = Object.keys(pkg.dependencies);
+const input = "dist-esm/lib/index.js";
+const production = process.env.NODE_ENV === "production";
+
+export function nodeConfig(test = false) {
+  const externalNodeBuiltins = ["events", "util"];
+  const baseConfig = {
+    input: input,
+    external: depNames.concat(externalNodeBuiltins),
+    output: { file: "dist/index.js", format: "cjs", sourcemap: true },
+    plugins: [
+      sourcemaps(),
+      replace({
+        delimiters: ["", ""],
+        values: {
+          // replace dynamic checks with if (true) since this is for node only.
+          // Allows rollup's dead code elimination to be more aggressive.
+          "if (isNode)": "if (true)"
+        }
+      }),
+      nodeResolve({ preferBuiltins: true }),
+      cjs(),
+      json()
+    ]
+  };
+
+  baseConfig.external.push("crypto");
+
+  if (test) {
+    // entry point is every test file
+    baseConfig.input = "dist-esm/tests/**/*.spec.js";
+    baseConfig.plugins.unshift(multiEntry({ exports: false }));
+
+    // different output file
+    baseConfig.output.file = "test-dist/index.js";
+
+    // mark assert as external
+    baseConfig.external.push("assert", "fs", "path", "os", "tty", "child_process");
+
+    baseConfig.onwarn = (warning) => {
+      if (warning.code === "THIS_IS_UNDEFINED") {
```

Comment:
Can you check if this custom onwarn is needed for this library? It may not be, depending on what dependencies are present...

Reply:
Also need to make sure any circular dep warnings aren't fatal by seeing if the resulting package works (in case you haven't yet :))

Reply:
I copied these from servicebus. It turns out that neither of these two are needed.

Reply:
Actually `chai` has some circular dep warnings but it is test only so we should be good.